components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/api/impl/BusinessRulesApiServiceImpl.java

```diff
@@ -166,23 +166,64 @@ public Response getTemplateGroups() throws NotFoundException {
 
     @Override
     public Response loadBusinessRule(String businessRuleInstanceID
- ) throws NotFoundException {
+    ) throws NotFoundException {
         TemplateManagerService templateManagerService = TemplateManagerInstance.getInstance();
         BusinessRule businessRule = templateManagerService.loadBusinessRule(businessRuleInstanceID);
         Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
         return Response.ok().entity(gson.toJson(businessRule)).build();
     }
+
     @Override
     public Response redeployBusinessRule(String businessRuleInstanceID
- ) throws NotFoundException {
+    ) throws NotFoundException {
+        TemplateManagerService templateManagerService = new TemplateManagerService();
+        int status;
+        status = templateManagerService.redeployBusinessRule(businessRuleInstanceID);
 
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        switch (status) {
+            case TemplateManagerConstants.SAVE_SUCCESSFUL_NOT_DEPLOYED:
+                return Response.ok().status(200).build();
+            case TemplateManagerConstants.SAVE_SUCCESSFUL_DEPLOYMENT_SUCCESSFUL:
+                return Response.ok().status(201).build();
+            case TemplateManagerConstants.SAVE_SUCCESSFUL_PARTIALLY_DEPLOYED:
+                return Response.ok().status(501).build();
+            default:
+                return Response.ok().status(500).build();
+        }
     }
+
     @Override
     public Response updateBusinessRule(String businessRule
-, String businessRuleInstanceID, Boolean deploy
- ) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+           String businessRuleInstanceID, Boolean deploy
+    ) throws NotFoundException {
+        TemplateManagerService templateManagerService = new TemplateManagerService();
+        Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
+        JsonObject businessRuleJson = gson.fromJson(businessRule, JsonObject.class);
+        int status;
+
+        if (businessRuleJson.get("type").toString().equals("\"" + TemplateManagerConstants.BUSINESS_RULE_TYPE_TEMPLATE
```

Comment:
Switch the sides of equals. And, make sure `businessRuleJson.get("type").toString()` doesn't cause any NPE.

Reply:
Fixed by b14dd18f396f799bd32670842869feb2475e778e